rp2040-hal/src/pio.rs

```diff
@@ -453,6 +455,7 @@ pub type PIO1SM3 = (PIO1, SM3);
 
 impl<P: PIOExt, SM: StateMachineIndex> ValidStateMachine for (P, SM) {
     type PIO = P;
+    type SM = SM;
```

Comment:
Perhaps call the type something like `SMIdx`? This `type SM = SM` looks a little bit strange, and `SM` isn't really referring to a specific state machine, only the full tuple `(P, SM)` does.